GoogleForCreators/web-stories-wp
# Add click detection to multi-selection Moveable

## Summary

Adds click detection to multi-selection moveable so that it would:
- Consider tiny drags / 0px drags as clicks instead
- Handling this click to either select the element or toggle its selection
- Clear selection if the click wasn't on any element.


## Relevant Technical Choices

Handling this is within Moveable multi-selection since Moveable takes over the mouse-up events handling in ending group events (`onGroupDragEnd` is relevant here).

This allows us conveniently use the clicked targets directly instead of having to calculate the click coordinates and compare these with the element coordinates.



## To-do




## User-facing changes

- The user can now toggle elements in multi-selection with holding down Shift key, and also select just one element in multi-selection


## Testing Instructions
- Add some elements and select them for multi-selection.

- Verify that the user can click on one of the elements in the selection and this will select it
- Verify that if the user clicks on empty area within the selection box, it will clear the selection
- Verify that the user can toggle elements in multi-selection by clicking + holding down the Shift key.


---



Fixes #5
